pandas/_testing/contexts.py

```diff
@@ -205,18 +205,24 @@ def use_numexpr(use, min_elements=None) -> Generator[None, None, None]:
         set_option("compute.use_numexpr", olduse)
 
 
-def raises_chained_assignment_error():
-    if PYPY:
+def raises_chained_assignment_error(extra_warnings=(), extra_match=()):
```

Comment:
here I'm giving `raises_chained_assignment_error` the ability to catch extra warnings (as this will be used later in https://github.com/pandas-dev/pandas/pull/50626)